### PR TITLE
Second lab

### PR DESCRIPTION
**Pull request by flatropw**
add factorial and square tasks

### CODE REVIEW

File: square.go [hunk 1]
```diff
@@ -1 +1,38 @@
...
+}
+
+type Square struct {
+	start Point
+	a     uint
+}
+
+func (s Square) End() Point {
+	/* some hardcoded stuff
+	i used the minus sign in calculation of y axis because in the attached screenshot start.y lower then end.y (???)
```
Comment (flatropw) [inline square.go:18]: Pay attention to minus sign. I made the right decision?
image

Comment (burov) [reply square.go:18]: Yep, end.y - start.y - a

File: square.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+	*/
+	return Point{s.start.x + int(s.a), s.start.y - int(s.a)}
+}
+
+func (s Square) Perimeter() uint {
+	const squareEdgesCount = 4
```
Comment (burov) [inline square.go]: You can use just magic number there as a contact it obvious what "4" means in scope of this function

File: square.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+	return s.a * squareEdgesCount
+}
+
+func (s Square) Area() uint {
+	return uint(math.Pow(float64(s.a), float64(2)))
```
Comment (burov) [inline square.go]: For me s.a * s.a, looks more obvious try to avoid redundant overcomplicating

**Review by burov — APPROVED** [review]
Well done:)

File: factorial.go [hunk 4]
```diff
@@ -0,0 +1,23 @@
+package main
+
+import (
+	"fmt"
+)
+
+func calculateFactorial(i uint) (result uint) {
```
Comment (imatsiash-ib) [inline factorial.go:7]: minor: according to the task description this function should have `factorial` name. No need to fix it here, just for the future.

**Review by imatsiash-ib — APPROVED** [review]

**Pull request closed without merging**